internal/services/recoveryservices/site_recovery_vmware_replicated_vm_resource.go

```diff
@@ -0,0 +1,1204 @@
+								DiskId:         *diskRaw.DiskId,
+								TargetDiskType: string(*diskRaw.DiskType),
+							}
+							if diskRaw.DiskEncryptionSetId != nil {
+								diskModel.TargetDiskEncryptionSetId = *diskRaw.DiskEncryptionSetId
+							}
+							diskOutputs = append(diskOutputs, diskModel)
+						}
+						state.DiskToInclude = diskOutputs
+					}
+
+					networkInterfaceModel := NetworkInterfaceModel{}
+					networkInterfaceModel.TargetStaticIp = pointer.From(inMageRcm.PrimaryNicIPAddress)
+
+					if inMageRcm.VMNics != nil {
+						nicsOutput := make([]NetworkInterfaceModel, 0)
+						for _, nic := range *inMageRcm.VMNics {
+							nicModel := NetworkInterfaceModel{}
+							nicModel.TargetStaticIp = pointer.From(nic.TargetIPAddress)
+							nicModel.TargetSubnetName = pointer.From(nic.TargetSubnetName)
+							nicModel.TestSubnetName = pointer.From(nic.TestSubnetName)
+							nicModel.SourceMacAddress = pointer.From(nic.NicId)
+							nicModel.IsPrimary = pointer.From(nic.IsPrimaryNic) == "true"
+
+							nicsOutput = append(nicsOutput, nicModel)
+						}
+						state.NetworkInterface = nicsOutput
+					}
+				}
+			}
+
+			var plan SiteRecoveryReplicatedVmVMwareModel
+			if err := metadata.Decode(&plan); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (s VMWareReplicatedVmResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 90 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			id, err := replicationprotecteditems.ParseReplicationProtectedItemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := metadata.Client.RecoveryServices.ReplicationProtectedItemsClient
+
+			disableProtectionReason := replicationprotecteditems.DisableProtectionReasonNotSpecified
+
+			disableProtectionInput := replicationprotecteditems.DisableProtectionInput{
+				Properties: replicationprotecteditems.DisableProtectionInputProperties{
+					DisableProtectionReason: &disableProtectionReason,
+					// It's a workaround for https://github.com/hashicorp/pandora/issues/1864
+					ReplicationProviderInput: &siterecovery.DisableProtectionProviderSpecificInput{
+						InstanceType: siterecovery.InstanceTypeDisableProtectionProviderSpecificInput,
+					},
+				},
+			}
+
+			err = client.DeleteThenPoll(ctx, *id, disableProtectionInput)
+			if err != nil {
+				return fmt.Errorf("deleting %s : %+v", id.String(), err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func resourceSiteRecoveryReplicatedVmVMWareClassicUpdateInternal(ctx context.Context, metadata sdk.ResourceMetaData) error {
+	client := metadata.Client.RecoveryServices.ReplicationProtectedItemsClient
+	containerClient := metadata.Client.RecoveryServices.ProtectionContainerClient
+	// We are only allowed to update the configuration once the VM is fully protected
+	state, err := waitForVmwareReplicationToBeHealthy(ctx, metadata)
```

Comment:
Should we move this to the end of `Create` (after pollilng) and `Update` (i.e. this function)?